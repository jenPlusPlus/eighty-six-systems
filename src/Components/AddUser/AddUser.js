import React, { Component } from 'react';
// import { Route, BrowserRouter } from 'react-router-dom';
import firebase from './../../firebase.js';

class AddUser extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      loginCode: ''
    };
  }
  componentDidMount() {
    const usersRef = firebase.database().ref('users');
    usersRef.on('value', (snapshot) => {
      let users = snapshot.val();
      for (let user in users) {
        console.log('user: ', users[user]);
      }
    });
  }

  updateState(key, event) {
    this.setState({[key]: event.target.value});
  }

  handleSubmit(event) {
    event.preventDefault();
    const usersRef = firebase.database().ref('users');
    const user = {
      name: this.state.name,
      loginCode: this.state.loginCode,
      tables: []
    };
    usersRef.push(user);

    this.setState({
      name: '',
      loginCode: ''
    });
  }

  render() {
      return (
      <div className="AddUser">
        <input type='text'
          placeholder='Name'
          value={this.state.name}
          onChange={(event)=>this.updateState('name', event)}/>
        <input type='text'
          placeholder='Login Code'
          value={this.state.loginCode}
          onChange={(event)=>this.updateState('loginCode', event)} />
        <button onClick={(event)=> this.handleSubmit(event)}>Submit</button>
      </div>
    );
  }
}

export default AddUser;
