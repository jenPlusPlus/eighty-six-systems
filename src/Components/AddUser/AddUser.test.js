import { shallow } from 'enzyme';
import AddUser from './AddUser';
import ReactDOM from 'react-dom';
import React from 'react';

describe('AddUser snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<AddUser />);

    expect(wrapper).toMatchSnapshot();
  });
});
