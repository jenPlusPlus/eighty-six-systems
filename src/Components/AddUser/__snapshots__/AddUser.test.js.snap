// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
/>,
        <input
          onChange={[Function]}
          placeholder="Login Code"
          type="text"
          value=""
/>,
        <button
          onClick={[Function]}
>
          Submit
</button>,
      ],
      "className": "AddUser",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Login Code",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
/>,
          <input
            onChange={[Function]}
            placeholder="Login Code"
            type="text"
            value=""
/>,
          <button
            onClick={[Function]}
>
            Submit
</button>,
        ],
        "className": "AddUser",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Login Code",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
